packages/react-native/Libraries/LinkingIOS/React-RCTLinking.podspec

```diff
@@ -28,7 +28,10 @@ header_search_paths = [
 if ENV["USE_FRAMEWORKS"]
   header_search_paths = header_search_paths.concat([
     "\"$(PODS_CONFIGURATION_BUILD_DIR)/ReactCommon/ReactCommon.framework/Headers/react/nativemodule/core\"",
-    "\"$(PODS_CONFIGURATION_BUILD_DIR)/React-NativeModulesApple/React_NativeModulesApple.framework/Headers\""
+    "\"$(PODS_CONFIGURATION_BUILD_DIR)/React-NativeModulesApple/React_NativeModulesApple.framework/Headers\"",
+    "\"$(PODS_CONFIGURATION_BUILD_DIR)/React-NativeModulesApple/React_NativeModulesApple.framework/Headers/build/generated/ios\"",
+    "\"$(PODS_CONFIGURATION_BUILD_DIR)/React-Codegen/React_Codegen.framework/Headers/build/generated/ios\"",
```

Comment:
What is generating code inside these folders? The default setup for Codegen does not do that.

Reply:
I'm actually not sure, that's a react-native iOS build detail beyond my understanding. 

To make this patch I repeatedly compiled the simple reproduction here https://github.com/evelant/rn072-ios-frameworks-static-failure and noted which pod and headers it was failing on each time. I located the missing header files in the build output and added those locations to the search path. 

After repeating the above enough times all the headers resolved correctly and the project compiled successfully.